scripts/check_go_lint.sh

```diff
@@ -3,38 +3,37 @@
 set -eu
 
 if [ $# -eq 0 ]; then
-  echo "$(basename $0) <go/root/dirs>" >&2
+  echo "$(basename "$0") <go/root/dirs>" >&2
   exit 1
 fi
 
 GOLINT=${GOLINT:-golangci-lint}
-ARGS=""
-ROOTS="$@"
+ROOTS=("$@")
 FAILED=
 GOROOTS=
 
 # only run linter in the root of go modules
-for root in ${ROOTS}; do
-  NEWROOTS=$(find ${root} -name .git -prune -o -name 'go.mod' -printf '%h\n' | sort -u | xargs)
-  if [ -n "${NEWROOTS}" ]; then
-    GOROOTS="${NEWROOTS} ${GOROOTS}"
+for root in "${ROOTS[@]}"; do
+  NEWROOTS=$(find "$root" -name .git -prune -o -name 'go.mod' -printf '%h\n' | sort -u | xargs)
+  if [ -n "$NEWROOTS" ]; then
+    GOROOTS="$NEWROOTS $GOROOTS"
```

Comment:
You used arrays elsewhere. It would also allow this to be space safe. Which it currently is not.